packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -679,11 +675,27 @@ export abstract class FluidDataStoreContext
 		this.channel!.setConnectionState(connected, clientId);
 	}
 
-	public notifyReadOnlyState(readonly: boolean): void {
+	public notifyReadOnlyState(): void {
 		this.verifyNotClosed("notifyReadOnlyState", false /* checkTombstone */);
 
-		this.channel?.notifyReadOnlyState?.(readonly);
-		this._contextDeltaManagerProxy.setReadonly(readonly);
+		// These two calls achieve the same purpose, and are both needed for a time for back compat
+		this.channel?.notifyReadOnlyState?.(this.isReadOnly());
+		this._contextDeltaManagerProxy.emitReadonly();
+	}
+
+	/**
+	 * Updates the readonly state of the data store based on the staging mode.
+	 *
+	 * @param staging - A boolean indicating whether the container is in staging mode.
```

Comment:
I'm having a big debate with myself about these `notifyFoo` params. You got rid of it for readonly above, to streamline the logic through `this.isReadOnly`. Makes sense. So can we add `IFluidParentContextPrivate.inStagingMode` and drop this param to match?

There are places where we've "lied" through an event parameter like this (@WillieHabi is working on untangling one between Container/ContainerRuntime), but it's only lead to confusion, it seems like an invariant worth holding (or just deduping like I'm promoting here) that the parameter matches what the getter says.

Reply:
For now, consistency seems worth prioritizing, so maybe add back the param for `notifyReadonlyState`? Even the `channel.notifyReadonlyState` call takes the parameter at the moment.

Reply:
i don't consider this a lie. we now support datastores being readonly when the container is not. the lie you talk about couples readonly and connected states in a way that can not be disambiguated by the client.

Reply:
this makes readonly more like attach state, which is really a per datastore state which can align with the container state, but doesn't need to.

Reply:
i'm also not particularly concerned about the api here, as it is all internal, and all within a single layer, so it is easy to change.

Reply:
Yeah it's fine.  Agreed there's no lie here, that's not what I was saying.